src/helpers/DataRequirementHelpers.ts

```diff
@@ -35,22 +37,82 @@ export function flattenFilters(filter: AnyFilter): AnyFilter[] {
  * @param filter the filter to translate
  * @returns codeFilter to be put on the DataRequirement
  */
-export function generateDetailedCodeFilter(filter: EqualsFilter | InFilter): R4.IDataRequirement_CodeFilter | null {
+export function generateDetailedCodeFilter(filter: EqualsFilter | InFilter, dataType: string): R4.IDataRequirement_CodeFilter | null {
```

Comment:
Suggest making the `dataType` argument optional here, that way some of the existing tests should hopefully pass as is when this is omitted.

This will require you to change the logic a bit when you actually call the `codeLookup` function, as you only want to do this when `dataType` is defined. 

Best approach for that is probably to initialize the `system` variable to be nothing first, and then only set its value inside an `if` for `if(dataType)`